dot/types/babe.go

```diff
@@ -10,6 +10,18 @@ import (
 // RandomnessLength is the length of the epoch randomness (32 bytes)
 const RandomnessLength = 32
 
+// AllowedSlots tells in what ways a slot can be claimed.
+type AllowedSlots byte
+
+const (
+	// PrimarySlots only allows primary slots.
+	PrimarySlots AllowedSlots = iota
+	// PrimaryAndSecondaryPlainSlots allow primary and secondary plain slots.
+	PrimaryAndSecondaryPlainSlots
+	// PrimaryAndSecondaryVRFSlots allows primary and secondary VRF slots.
+	PrimaryAndSecondaryVRFSlots
+)
+
```

Comment:
is this the same as substrate? can you add a link if so?

Reply:
done